apis/python/src/tiledbsoma/util_tiledb.py

```diff
@@ -150,11 +139,21 @@ def open_group(
         mode: options.OpenMode,
         context: SOMATileDBContext,
     ) -> "ReadWriteHandle[tiledb.Group]":
+        return cls._open(tiledb.Group, uri, mode, context)
+
+    @classmethod
+    def _open(
+        cls,
+        tiledb_cls: Callable[..., _TDBO],
+        uri: str,
+        mode: options.OpenMode,
+        context: SOMATileDBContext,
+    ) -> "ReadWriteHandle[_TDBO]":
         try:
-            rd = tiledb.Group(uri, "r", ctx=context.tiledb_ctx)
+            rd = tiledb_cls(uri, "r", ctx=context.tiledb_ctx)
             try:
```

Comment:
Might be nice to let an `ExitStack` manage unwinding the handles on exception paths here and in `close()`, esp if we're soon going to have a third thing `SOMAReader` to unwind.

Reply:
This is going to end up further changing pretty soon (better handling of write-only handles) but I will see if that makes sense after those revisions.